src/main/java/se/heinrisch/talkclient/TalkClient.java

```diff
@@ -167,9 +180,11 @@ public void onResult(DataItemBuffer dataItems) {
                     final DataMapItem dataMapItem = DataMapItem.fromDataItem(dataItems.get(i));
                     if (dataMapItem.getUri().getPath().equals(path)) {
                         mTalkDataAdapter.onRecieveResult(dataMapItem.getDataMap());
-                        continue;
+                        dataItems.release();
+                        return;
```

Comment:
The dataItems should be released when they are processed so the question is where that should be done. The idea behind doing it in this function is that it wont be forgotten and if forces the caller to process the dataitem at once instead of storing the data item.

The reasoning behind `mTalkDataAdapter.onRecieveResult(new DataMap())` is that you should always get a callback on getDataItems. But that might be a little confusing.